site/src/components/Avatar/Avatar.tsx

```diff
@@ -11,26 +11,26 @@ export type AvatarProps = MuiAvatarProps & {
 };
 
 const sizeStyles = {
-  sm: (theme: Theme) => ({
+  sm: (theme) => ({
     width: theme.spacing(3),
     height: theme.spacing(3),
     fontSize: theme.spacing(1.5),
   }),
   md: {},
-  xl: (theme: Theme) => ({
+  xl: (theme) => ({
     width: theme.spacing(6),
     height: theme.spacing(6),
     fontSize: theme.spacing(3),
   }),
-};
+} satisfies Record<string, Interpolation<Theme>>;
```

Comment:
that's what is cool about `satisfies`, it doesn't widen the type, it makes sure that `T extends Record<...>`. if you added an incorrect key here, it would still cause a type error at the lookup, because TS still knows the actual keys the object has.

Reply:
So, you're right that it doesn't widen the type, but I guess the way I look at it is that right now, we have a narrow type, but wide type-checking, when ideally both would be as narrow as possible.

So for this example specifically, there's nothing requiring that all sizes be present in the record – the key for type-checking is any arbitrary string, so there's no protection against typos in the runtime value's key names, or accidentally forgetting a size

Like, if we decide to add a "lg" size down the line, there aren't any protections to make sure we don't forget to add a matching style. But with the record being keyed by all members of the union, TypeScript will complain if there are keys that don't match, or if we forget to include all the keys

Reply:
there are! the lookup will fail, we don't need a more stringent `satisfies` to catch that. it'd just make this type harder to write.

Reply:
Oh wait, yeah, you're right. I was worried because we don't have the `noUncheckedIndexedAccess` compiler setting turned on, which can cause issues with trying to read properties from a record, but because it technically isn't a record, and just has the type-checking of one, it's safe.

We will get errors when trying to consume a property that doesn't exist

Reply:
yup, exactly!